### DISCUSSION TITLE
useRouter not working as properly

### DISCUSSION THREAD

**Discussion by ChaowuBest** [vercel/next.js#52253] [Help] [answered]
### Summary

I am using nextjs dynamic router as my product page, the folder structure is pages/products/[product].tsx. I tried to get the real path, when I try to use Link component redirect to this page, sometimes this line(  const router = useRouter()) not invoked, but when I refresh this page, it works but always not get the real path. 

Question: what should I do to get real path for dynamic route page?

### Additional information

```js
//This is how I navigate to dynamic router page: 
 <Link href={product.url}   // href ="/products/XX-XXXX"
                                      className={"group"}>
                                    <Image width={320} height={320} alt={""} src={product.imageUrl}
                                           className={"object-center"}></Image>
                                    <span className={"block group-hover:underline"}>{product.category}</span>
                                    <span className={"block group-hover:underline"}>{product.name}</span>
                                    <span
                                        className={"after:inline-block after:ml-1 after:content-['SEK']"}>{product.price}</span>
                                </Link>

//This is how I process dynamic router page: 
'use client'
import {useRouter} from 'next/navigation';

export default function Product() {
    const router = useRouter()
    //get path
    //process page
    return (
        <div className={"fixed inset-0 bg-red-200"}>
        </div>
    )
}
```

**Comment by icyJoseph** [comment, marked as answer]
If you have `pages/products/[product].tsx` as file name, then you are using the Pages Router, and there the router comes from `next/router`, not `next/navigation`

The `next/navigation` router doesn't contain any info about path, what so ever, it only has navigation methods.

**Reply by ChaowuBest** [reply]
Sorry for late reply, finally figured out, thanks
